src/OrchardCore/OrchardCore.Abstractions/Json/Dynamic/JsonDynamicObject.cs

```diff
@@ -80,6 +80,13 @@ public override bool TryInvokeMember(InvokeMemberBinder binder, object?[]? args,
         return true;
     }
 
+    public bool Remove(string key) => _jsonObject.Remove(key);
```

Comment:
I like the idea of having the remove method, but it should take `_dictionary` into consideration. Right now if you access a key with `GetValue` and then remove it, then accessing it again with `GetValue` will still return the old property. In other words a test such as this should pass:

```csharp
        [Fact]
        public void RemovingPropertyShouldWork()
        {
            var contentItem = new ContentItem();
            contentItem.GetOrCreate<MyPart>();
            contentItem.Alter<MyPart>(x => x.Text = "test");

            JsonDynamicObject content = contentItem.Content;
            Assert.Null(content.GetValue("not real property")); // Properties that don't exist return null.
            Assert.NotNull(content.GetValue(nameof(MyPart))); // Right now this property exists.

            content.Remove(nameof(MyPart));
            Assert.Null(content.GetValue(nameof(MyPart)));
        }
```

This is what you see if you break at the last line of the above test:
![image](https://github.com/OrchardCMS/OrchardCore/assets/4007293/b44b57d3-033a-4749-bd0a-45591db40c2f)


The same applies to `JsonDynamicArray` as well.